subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -986,8 +999,16 @@ int bt_mesh_sensor_cli_series_entries_get(
 		col_format = bt_mesh_sensor_column_format_get(sensor);
 
 		if (!col_format) {
-			net_buf_simple_add_le16(&msg, range->start.val1);
-			net_buf_simple_add_le16(&msg, range->end.val1);
+			if (range->start.val1 < 0xFF && range->end.val1 < 0xFF) {
+				net_buf_simple_add_u8(&msg, range->start.val1);
+				net_buf_simple_add_u8(&msg, range->end.val1);
+			} else if (range->start.val1 < 0xFFFF && range->end.val1 < 0xFFFF) {
+				net_buf_simple_add_le16(&msg, range->start.val1);
+				net_buf_simple_add_le16(&msg, range->end.val1);
+			} else {
+				net_buf_simple_add_le24(&msg, range->start.val1);
+				net_buf_simple_add_le24(&msg, range->end.val1);
+			}
```

Comment:
MshMDLd1.1r07, 4.2.17 Sensor Series Get (emphasis mine):

> [...]
> 
> If present, the Raw Value A1 field identifies a starting column of a sensor’s series within an element (see Section 4.1.5.2) or identifies a **2-octet starting index** of the measured value counting backwards from the most recent value (represented as index 0x0000) for the device properties that have only one or two fields in the Property Value.
> 
> If present, the Raw Value A2 field identifies an ending column of a sensor’s series within an element (see Section 4.1.5.2) or identifies a **2-octet last index** of the measured value counting backwards from the most recent value for the device properties that have only one or two fields in the Property Value.

Reply:
Oh, I was looking for this but didn't find that, thank you! I saw PTS waiting for value 1 octet value, but that looks like PTS issue; I'll remove it

Reply:
Hmm, one problem with this: this is not a public (released) spec, this is public repo, and issue is with regular (released) PTS version. I'm not sure how I should handle this.
> The Raw Value X1 field identifies a starting column of a sensor’s series within an element (see
Section 4.1.5.2).
The Raw Value X2 field identifies an ending column of a sensor’s series within an element (see
Section 4.1.5.2).

>4.1.5.2
 Sensor Raw Value X
The Sensor Raw Value X field has a size and representation defined by the Sensor Property ID and
represents the left corner of the column on the X axis.

This is what MMDLv1.0.1 states, nothing more.

Reply:
Right. Meaning that where the Raw Value X size was *not* clear based on the Property ID (i.e. for single channel sensors) the meaning was unclear. This is why the spec has been clarified. But as far as I can tell, the Raw Value X1 and Raw Value X2 as well as what Property ID to use is selected by us using MMI right? Meaning we could just use a sensor type with three channels, which have clearly defined expected behavior for both the old and revised spec?